servers.tf

```diff
@@ -63,52 +69,111 @@ resource "google_compute_instance_template" "vault" {
   depends_on = [google_project_service.service]
 }
 
+############################
+## Internal Load Balancer ##
+############################
+
+resource "google_compute_health_check" "vault_internal" {
+  count   = local.use_internal_lb ? 1 : 0
+  project = var.project_id
+  name    = "vault-health-internal"
+
+  check_interval_sec  = 15
+  timeout_sec         = 5
+  healthy_threshold   = 2
+  unhealthy_threshold = 2
+
+  http_health_check {
+    port = var.vault_proxy_port
+  }
+
+  depends_on = [google_project_service.service]
+}
+
+resource "google_compute_region_backend_service" "vault_internal" {
+  count         = local.use_internal_lb ? 1 : 0
+  name          = "vault-backend-service"
+  region        = var.region
+  health_checks = ["${google_compute_health_check.vault_internal[0].self_link}"]
+  backend {
+    group = google_compute_region_instance_group_manager.vault.instance_group
+  }
+}
+
+# Forward internal traffic to the backend service
+resource "google_compute_forwarding_rule" "vault_internal" {
+  provider = google-beta
+  count    = local.use_internal_lb ? 1 : 0
+  project  = var.project_id
+
+  name                  = "vault-internal"
+  region                = var.region
+  ip_protocol           = "TCP"
+  ip_address            = var.internal_lb_ip
+  load_balancing_scheme = var.load_balancing_scheme
+  network_tier          = "PREMIUM"
+  allow_global_access   = true # BETA
```

Comment:
If this used to be a beta feature, it's not anymore. You can remove `google-beta` from this now. Actually, I don't think you need `google-beta` anymore as a provider now.

Reply:
It was beta at the time of writing this PR. Updated now.